### PR TITLE
Reused Reportbydate component for HIV, Summary of All tests and Rejection Reports

### PR DESCRIPTION
**Pull request by konavivekramakrishna**
This pr removes the redundant code for HIV, Summary of All tests and Rejection report and reuses the `reportByDate` component 

and also adds validation by preventing future date and not allowing for empty date fields for generating report

After : 

image

Complete workflow : 

https://www.loom.com/share/68c997a40d754ebea25479212804c4e5?sid=fc9ffb15-5f52-43c3-9aee-e4cb41084c96

### CODE REVIEW

File: frontend/src/components/Reports/study/common/ReportByDate.js [hunk 1]
```diff
@@ -53,13 +47,29 @@ const ReportByDate = (props) => {
...
+        }),
+      });
+      return;
+    }
+
+    setReportFormValues({
+      ...reportFormValues,
+      error: "",
+    })
+
```
Comment (manishjha-04) [inline frontend/src/components/Reports/study/common/ReportByDate.js:65]: Hey @konavivekramakrishna, just wanted to offer a suggestion. You could add a handleChange function that clears any errors. In the video on Loom, I noticed that even if the user selects a date, the error message remains visible

Comment (manishjha-04) [reply frontend/src/components/Reports/study/common/ReportByDate.js:65]: Something like `const handleInputChange = (event) => {
    setErrors({});
  };`
aligning with best practices for usability and error handling

File: frontend/src/components/Reports/Routine.js [hunk 2]
```diff
@@ -129,8 +127,7 @@ export const RoutineReportsMenu = {
       SideNavMenuItem: [
         {
           link:
-            config.serverBaseUrl +
-            "/Report?type=patient&report=CISampleRoutineExport",
+            "/CISampleRoutineExport?type=patient&report=CISampleRoutineExport",
```
Comment (manishjha-04) [inline frontend/src/components/Reports/Routine.js:130]:
```suggestion
            "/RoutineReport?type=patient&report=CISampleRoutineExport",
```

**Review by mherman22 — COMMENTED** [review]
Thanks @konavivekramakrishna , please make sure to main these breadcrumbs. Because they make navigating back to the Routine Reports index page less painful. SOmething that was added in https://github.com/I-TECH-UW/OpenELIS-Global-2/commit/eafe6447598ec7978ad7b07915b1963fd6301207. And in your demo video, remember to go through all Routine Reports so that we don't leave anything unfixed this time-round.

Otherwise it's great to try to make some stuff reusable to avoid duplication (which is evident in the entire codebase)

## example
Screenshot from 2024-03-19 15-04-02

**Comment by codewithunknown** [comment]
@mherman22 
image

How about this breadcrumbs ?

**Comment by mherman22** [comment]
> @mherman22
> 
> image
> 
> How about this ?

Awesome 😁

**Review by mherman22 — APPROVED** [review]
LGTM 🥳

cc: @mozzy11

**Pull request merged**